maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -366,20 +366,21 @@ private void afterSessionEnd( Collection<MavenProject> projects, MavenSession se
 
     private void saveResumptionDataWhenApplicable( MavenExecutionResult result, MavenSession session )
     {
-        List<LifecycleExecutionException> lifecycleExecutionExceptions = result.getExceptions().stream()
+        long lifecycleExecutionExceptionCount = result.getExceptions().stream()
                 .filter( LifecycleExecutionException.class::isInstance )
-                .map( LifecycleExecutionException.class::cast )
-                .collect( Collectors.toList() );
+                .count();
```

Comment:
Why do we need a stream to count elements?

Reply:
We think this is the most readable way of counting those specific exceptions. Rewriting it in a for-loop could potentially save some resources (if javac does not already optimize the stream for us). We usually go for easy-to-read code first, and optimize when needed.
That said, I am not sure if the Maven codebase has a different strategy? Maybe @rfscholte or someone else can chip in?

Reply:
The next codeblock needs to know if there were lifecycleExecutionException. That said, the amount doesn't matter, a simple `boolean` would be sufficient. A count needs to loop over all exceptions, whereas replacing `filter` with `anyMatch` would be the optimized implementation.

Reply:
Ah yes, of course, that makes perfect sense!